cpp/src/copying/contiguous_split.cu

```diff
@@ -142,12 +142,12 @@ __device__ void copy_buffer(uint8_t* __restrict__ dst,
                             int num_rows,
                             size_type* valid_count)
 {
-  src += (src_row_index * element_size);
+  src += (static_cast<size_t>(src_row_index) * static_cast<size_t>(element_size));
```

Comment:
Why not just make the input argument type `std::size_t`? You're not saving anything by doing this.